### PR TITLE
IA-448: Forms

### PR DESCRIPTION
**Pull request by beygorghor**
- edit form on a detail page
- add form versions array to detail page + adapt API
- add start and end period to FormVersion, as a string for now + adapt API
- replace create/update logic into async/ await, not sure it's more clear, at least not worst
- raise unit test coverage to 100% into forms folder 

Screenshot 2021-04-08 at 00 56 30
Screenshot 2021-04-08 at 00 52 16

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/forms/config.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 import React from 'react';
 import moment from 'moment';
-import Grid from '@material-ui/core/Grid';
+import { Grid } from '@material-ui/core';
```
Comment (vinch) [inline hat/assets/js/apps/Iaso/domains/forms/config.js:3]: Why?

File: hat/assets/js/apps/Iaso/domains/forms/detail.js [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+                        },
+                        isUpdate,
+                    );
+                    isSaveSuccessful = true;
+                } catch (createVersionError) {
+                    // when creating form, if version creation fails, delete freshly created, version-less form
+                    if (!isUpdate) {
+                        try {
+                            await deleteForm(dispatch, savedFormData.id);
+                            console.log('Form deleted');
```
Comment (vinch) [inline hat/assets/js/apps/Iaso/domains/forms/detail.js:138]: What's your policy regarding console.log? Do you keep some of them in the codebase or is it for testing purposes?

File: hat/assets/js/apps/Iaso/domains/forms/config.js (hunk 1 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/forms/config.js:3]: used line 142

File: hat/assets/js/apps/Iaso/domains/forms/detail.js (hunk 2 above)
Comment (beygorghor) [reply hat/assets/js/apps/Iaso/domains/forms/detail.js:138]: nope, it was there before... i assume Pierre wanted to let it fail silently and keep a trace

**Pull request merged**